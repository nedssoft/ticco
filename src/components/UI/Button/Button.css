.Button {
  background-color: transparent;
  border: none;
  color: white;
  outline: none;
  cursor: pointer;
  font: inherit;
  padding: 10px;
  margin: 10px;
  font-weight: bold;
}

.Button:first-of-type {
  margin-left: 0;
  padding-left: 0;
}

.Success {
  color: #5C9210;
}

.Danger {
  color: #944317;
}
.Button:disabled {
  color: #ccc;
  cursor: not-allowed;
}
.btnDefault{
  padding: 10px 0 10px 0;
  text-align: center;
  line-height: 20px;
  background-color: #EE6860;
  letter-spacing: 0.3px;
  color: aliceblue;
  border: none;
  outline: none;
  font-size: 1em;
  font-weight: 600;
  text-decoration: none;
  cursor: pointer; 
}