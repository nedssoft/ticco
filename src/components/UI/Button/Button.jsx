import React from 'react'
import classes from './Button.css'

const Button = ({ clicked, children, btnType, disabled}) => {
  return (
    <button 
      onClick={clicked}
      disabled={disabled}
      className={[classes.Button, classes[btnType]].join(' ')}
    >
      { children }
    </button>
      
  )
}

export default Button
