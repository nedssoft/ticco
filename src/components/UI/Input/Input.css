.Input {
  width: 100%;
}
.Label {
  font-weight: bold;
  display: block;
  margin-bottom: 5px;
  box-sizing: border-box;
  text-align: left;
  width: 100%;
  margin-left: 10%;
}
.InputElement{
  outline: none;
  width: 80%;
  border: 1px solid #ccc;
  background: #fff;
  padding: 6px 10px;
  font: inherit;
  display: block;
  margin: 10px auto;
  font-size: 18px;
  border-radius: 3px;
}

.InputElement:focus {
  outline: none;
  background: #ccc;
}

.Invalid {
  border: 1px solid #bb4430;
  background-color: #bb4430;
}

.Valid {
  border: 1px solid rgb(106, 161, 106)
}
