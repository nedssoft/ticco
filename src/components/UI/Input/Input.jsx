import React from 'react'

import styles from './Input.css'


const Input = (props) => {
  const { 
    elementType, 
    elementConfig, 
    value, 
    label, 
    changed,
    valid,
    shouldValidate,
    touched
  } = props;
  const { id } = elementConfig
  let inputElement = null
  const inputClasses = [styles.InputElement]

  if (!valid && shouldValidate && touched) {
    inputClasses.push(styles.Invalid)
  }
  if (valid) {
    inputClasses.push(styles.Valid)
  }
  switch (elementType) {
    case 'input':
      inputElement = (
        <input 
          {...elementConfig}
          className={inputClasses.join(' ')}
          onChange={changed}
          value={value}
          id={id}
        />
      );
      break;
    case 'textarea':
      inputElement = (
        <textarea 
          {...elementConfig} 
          className={inputClasses.join(' ')} 
          onChange={changed}
          value={value}
          id={id}
        />
      )
      break;
      case 'select':
      inputElement = (
        <select 
          className={inputClasses.join(' ')} 
          onChange={changed}
          value={value}
          id={elementConfig.id}
          label={elementConfig.label}
        >
          {elementConfig.options.map(option => (
            <option value={option.value} key={option.value}>
              {option.displayValue}
            </option>
          ))} 
        </select>
      )
      break;
    default:
      inputElement = (
        <input 
          {...elementConfig} 
          className={inputClasses.join(' ')} 
          onChange={changed}
          value={value}
        />
      )
}
  return (
    <div className={styles.Input}>
      <label htmlFor={id} className={styles.Label}> {label}</label>
      { inputElement }
    </div>
  )
}

export default Input
