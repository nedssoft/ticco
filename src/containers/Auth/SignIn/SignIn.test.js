import React from 'react';
import { shallow } from 'enzyme'
import { SignIn } from './SignIn'
import Button from '../../../components/UI/Button/Button'
import Input from '../../../components/UI/Input/Input'

describe('Auth Components', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<SignIn />)
  })
 it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
 })
 it('should render form element', () => {
  expect(wrapper.find('form')).toHaveLength(1)
 })
 it('should render Button component', () => {
  expect(wrapper.find(Button)).toHaveLength(1)
 })
 it('should render Input component', () => {
  expect(wrapper.find(Input)).toHaveLength(2)
 })
})