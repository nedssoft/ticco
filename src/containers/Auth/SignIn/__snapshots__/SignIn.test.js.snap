// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Components should render correctly 1`] = `
<div
  className="Container"
>
  <div
    className="Auth"
  >
    <form
      onSubmit={[Function]}
    >
      <Input
        changed={[Function]}
        elementConfig={
          Object {
            "id": "email",
            "placeholder": "Your E-mail",
            "type": "email",
          }
        }
        elementType="input"
        key="email"
        label="Email"
        shouldValidate={true}
        valid={false}
        value=""
      />
      <Input
        changed={[Function]}
        elementConfig={
          Object {
            "id": "password",
            "placeholder": "Password",
            "type": "password",
          }
        }
        elementType="input"
        key="password"
        label="Password"
        shouldValidate={true}
        valid={false}
        value=""
      />
      <Button
        btnType="Submit"
        disabled={true}
      >
        Sign In
      </Button>
    </form>
    <div
      className="Links"
    >
      <Link
        to="/password-reset"
      >
         Forgot Password?
      </Link>
      <Link
        to="/signup"
      >
         Not Registered? Sign up
      </Link>
    </div>
  </div>
</div>
`;
